### PR TITLE
Display a recommended badge for all add-on types, and hide the featured badge for all add-on types

### LINKED ISSUE #8035: Display recommended bage for all add-on types
As currently implemented, the Recommended badge only appears for extensions. A decision was made to display it for all add-on types, not just extensions. This issue is about removing the restriction(s) that make the badge exclusive to extensions.

Note that this also changes the behaviour of the "Featured" badge when the "Recommended" feature is enabled. With this patch, no add-on types will display a "Featured" badge when the "Recommended" feature is enabled.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #8035

### CODE REVIEW

File: tests/unit/amo/components/TestAddonBadges.js [hunk 1]
```diff
@@ -136,27 +118,10 @@ describe(__filename, () => {
     expect(root.find(Badge)).toHaveProp('label', 'Featured Extension');
   });
 
-  it('displays a featured badge for a theme when the addon is featured and the recommended feature is on', () => {
+  it('does not display a badge for a featured addon when the recommended feature is on', () => {
```
Comment (kumar303) [inline tests/unit/amo/components/TestAddonBadges.js]: oof, I get it but this is confusing

**Comment by bobsilverberg** [comment]
> r+wc
> 
> Hmm, I feel like this layout could be better:
> 
> Screenshot 2019-05-22 15 27 51
> 
> Maybe the badge should be under the theme image, to the right of the title. However, this is not easy to do as the grid for this is [weird](https://github.com/mozilla/addons-frontend/blob/ae40a8fd9192fb7908a9cac3f87f8950d0bde8cd/src/amo/pages/Addon/styles.scss#L151).

Yes, this doesn't look great. @brassy- , do you have any advice as to where we should put the badge on the Theme detail page?

**Comment by brassy-** [comment]
@bobsilverberg Here two possible solutions

1. Under the theme image, on the same line of the title, right-aligned. 
description-badge jpg

2. Below the description
title-badge jpg

**Comment by bobsilverberg** [comment]
Thanks @brassy-. How about like this, between the author and the description:

Screenshot 2019-05-23 09 37 31

The reason I am suggesting that is because that is where all the other badges will currently appear for a theme, for example:

Screenshot 2019-05-23 09 39 32

**Comment by brassy-** [comment]
> Screenshot 2019-05-23 09 39 32

Can we have the badges on the same line, one next to the other?

**Comment by bobsilverberg** [comment]
> Can we have the badges on the same line, one next to the other?

As per our discussion on Slack, here are some samples:

This is how it would currently look with multiple badges with one on each line:
Screenshot 2019-05-27 12 16 09

Here is how the same would look with all badges on the same line:
Screenshot 2019-05-27 12 15 11

Here is how it would look on a smaller screen size, where all of the badges do not fit nicely on one line:
Screenshot 2019-05-27 12 15 46

Bear in mind that it would be pretty unusual to have multiple badges for a recommended add-on, as most badges are for reasons that would likely exclude an add-on from being recommended.

@brassy- What are your thoughts?

**Comment by bobsilverberg** [comment]
@brassy- Here are a couple of examples of what I came up with after our discussion. Putting multiple badges on a single line, but wrapping when they won't all fit. Using `12px` spacing between badges.

Screenshot 2019-05-27 12 48 05

Screenshot 2019-05-27 12 48 28

What do you think?

**Comment by brassy-** [comment]
Thank you! It looks great.

👍

**Comment by bobsilverberg** [comment]
Thanks @brassy- !

**Pull request merged**